codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAwsRuntimeConfig.java

```diff
@@ -95,8 +95,13 @@ public void addConfigInterfaceFields(
                     .write("serviceId?: string;\n");
         }
         if (isSigV4Service(settings, model)) {
-            writer.writeDocs("The AWS region to which this client will send requests or use as signingRegion")
-                    .write("region?: string | __Provider<string>;\n");
+            if (isAwsService(settings, model)) {
+                writer.writeDocs("The AWS region to which this client will send requests")
+                        .write("region?: string | __Provider<string>;\n");
+            } else {
+                writer.writeDocs("The AWS region to use as signing region for AWS Auth")
+                        .write("region?: string | __Provider<string>;\n");
+            }
```

Comment:
Writing the actual region can be outside of this block since it doesn't change.

EDIT: actually it seems like the rest of the changes in this PR serve to effectively make region required, so I guess you'd want to enforce that here

Reply:
I had the `if` to write a specific comment in each case.

It is not required in the sense that you don't need to provide it when instantiating the client. It is link for [AWS clients](https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-accessanalyzer/AccessAnalyzerClient.ts#L231) where it comes in to [ClientDefaultValues](https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-accessanalyzer/AccessAnalyzerClient.ts#L310) from [here](https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-accessanalyzer/runtimeConfig.ts#L32)

Reply:
Yes, but you can do that and still write the region separately. The methods in `TypeScriptWriter` return the object so you can chain calls like this, but it's not necessary.